main.py

```diff
@@ -648,6 +644,16 @@ async def scrape_channel_media(
                 )
             )
 
+    # Save all files if not in cache
+    async def dl_file(attachment: Attachment, attachment_filepath: str) -> None:
+        if not os.path.exists(attachment_filepath):
+            await attachment.save(attachment_filepath)
+
+    tasks = [
+        asyncio.create_task(dl_file(at, fp)) for _, at, fp in channel_media_attachments
```

Comment:
Do we actually need to use `asyncio.create_task` here? [`asyncio.wait`](https://docs.python.org/3/library/asyncio-task.html#asyncio.wait) states that it can act on any Awaitable object, which includes coroutines.

Reply:
passing coroutines directly to `async.wait` is [deprecated](https://docs.python.org/3/library/asyncio-task.html#asyncio.wait) and will be removed in Python 3.11 (see italicized last few lines in `async.wait` section), so we should stick to creating tasks.